tests/test_transformation/test_logcp_transformer.py

```diff
@@ -0,0 +1,155 @@
+import pandas as pd
+import pytest
+from sklearn.exceptions import NotFittedError
+
+from feature_engine.transformation import LogCpTransformer
+
+_age_inverse_transform = [20.0, 21.0, 19.0, 18.0]
```

Comment:
If we add one variable with values >0 and < than 1 we would catch the bug on having the test for <0 in the inverse_transform

Reply:
ok, so if removing the check in inverse transform, no need for testing this, right?

Reply:
it doesn't hurt to test in my opinion :)

Reply:
so if we add an "Score" variable here

```python
import pandas as pd
import numpy as np

data = {
        "Name": ["tom", "nick", "krish", "jack"],
        "City": ["London", "Manchester", "Liverpool", "Bristol"],
        "Age": [20, 21, 19, 18],
        "Marks": [0.9, 0.8, 0.7, 0.6],
        "Score": [0.15, 0.007, 0.32, 0.21],
        "dob": pd.date_range("2020-02-24", periods=4, freq="T"),
    }

df_vartypes = pd.DataFrame(data)
df_vartypes
```

what should we test for ?, considering we remove the check in `inverse_transform` 

```python 
# check contains zero or negative values
if (X[self.variables_] + self.C_ <= 0).any().any():
    raise ValueError(
        "Some variables contain zero or negative values after adding"
        + "constant C, can't apply log"
    )
```

Reply:
You could use the "Marks" variable directly. That one should have catched the error raising the variables should not be negative in the inverse transform.

Reply:
Oh, I understand the question now. Sorry. Applying the inverse_transform on a variable with values >-1, ensures the transformer works not only for variables >1 but also >0 and <1. And protects us automatically for introducing an error in the future. The error does not exist today, but we may accidentally introduce it in the future if we forget why we removed it, or if we modify some other masterclass from which this transformer inherits.

Reply:
yes, thanks for clarifying, this is an easy addition, might be similar to `test_inverse_transform`